### DISCUSSION TITLE
Is it normal that formulas are recalculated each time a file is saved?

### DISCUSSION THREAD

**Discussion by Krakor92** [mdelobelle/metadatamenu#185] [Q&A] [answered]
Hello, 

I noticed that when I put a console.log inside the formula field of a file, it prints in the devtools console every time a file is saved, even if the original file with the formula field is closed...

Is this intentional?

It bothers me a bit because I try to use long one-line ternary formulas in some of my classes, but it's just not usable right now because of that (every 5 keystrokes my CPU load has a huge spike).

**Comment by mdelobelle** [comment, marked as answer]
Yes that's the standard behaviour: unlike dataview, lookup and formula fields are recalculated and "printed" in the note. And since we don't know if a given change in one note has or hasn't an impact on a given formula, the plugin has to recalculate. Maybe could I add a setting to "not autoupdate" (just like for lookup fields)
in this case, the formula would be updated only when the note is opened again.

It that makes sense to you, could you open an issue for that?

**Reply by Krakor92** [reply]
Ok thanks, i just opened the [issue](https://github.com/mdelobelle/metadatamenu/issues/188)
